src/libraries/System.Text.Json/src/System/Text/Json/JsonException.cs

```diff
@@ -125,7 +125,7 @@ public override void GetObjectData(SerializationInfo info, StreamingContext cont
         /// <summary>
         /// The path within the JSON where the exception was encountered.
         /// </summary>
-        public string Path { get; internal set; }
+        public string Path { get; internal set; } = string.Empty;
```

Comment:
Did we consider making `Path` nullable?

Reply:
it was nullable first, there discussion about it, last suggestion from @bartonjs was to make it non null and default with emty string https://github.com/dotnet/runtime/pull/114#discussion_r348870828

Reply:
**OK, for now, let's make `Path` nullable to remain consistent with what the existing 3.1 behavior is.** I don't see a reason why it should be non-nullable. What is the benefit? Also, this change is causing the binary formatter test to break.

I don't think an empty Json Path string has any useful semantic meaning, so we could go either way here.

Previously, when the `Utf8JsonReader` threw a `JsonException`, the `Path` would be null. So, I don't know if we should conflate "path doesn't exist"/"path not set" with "empty path" (that said, the `JsonSerializer` doesn't return an empty path today, it always has at least "$").

We can revisit this later, if needed. This also means make the ctor params nullable and revert the change in `JsonReaderException.cs`